### PR TITLE
#9 add destination documentation

### PR DESCRIPTION
**Pull request by CAMOBAP**
- #9 

cc @ronaldtse

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +69,36 @@ By default, the action will try to compile/generate docs and upload them as page
...
+- `artifact` - Upload site artifact as zip archive
+  ```yml
+  ...
+  + uses: actions-mn/build-and-publish@main
+    with:
+      destination: artifact
+      agree-to-terms: true
+  ```
+
+> *Important note*. In case of `default` you must pass `token`, it's required to determine if pepository allow to GH Pages deployment
```
Comment (ronaldtse) [inline README.md:100]: "GH" => "GitHub"

File: action.yml [hunk 2]
```diff
@@ -2,8 +2,8 @@ name: 'build-and-publish'
 description: 'Perform metanorma site generate and upload pages artifact'
 inputs:
   token:
-    description: secrets.GITHUB_TOKEN
-    default: '.'
+    description: Required if destination='auto'. Just pass secrets.GITHUB_TOKEN
```
Comment (ronaldtse) [inline action.yml:5]: Wrap backticks around `secrets.GITHUB_TOKEN` ?

File: action.yml [hunk 3]
```diff
@@ -48,6 +48,11 @@ runs:
         GH_TOKEN: ${{ inputs.token }}
       shell: bash
       run: |
+        if [ "$GH_TOKEN" = "" ]; then
+          echo "Error! For destination=default you have to pass token input also."
```
Comment (ronaldtse) [inline action.yml:52]: Change message to: "Please provide the `secrets.GITHUB_TOKEN` as input for GitHub Pages deployment."

File: action.yml [hunk 4]
```diff
@@ -48,6 +48,11 @@ runs:
...
+          exit 1
+        fi
+
         if ! type gh > /dev/null; then
           curl -L https://github.com/cli/cli/releases/download/v2.24.3/gh_2.24.3_linux_amd64.tar.gz | tar xz --strip-components=2 -C /usr/local/bin/ gh_2.24.3_linux_amd64/bin/gh
```
Comment (ronaldtse) [inline action.yml:57]: Please move the remote bin path to an environment variable?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: action.yml (hunk 4 above)
Comment (CAMOBAP) [reply action.yml:57]: I think I will move gh setup to separate action

Comment (CAMOBAP) [reply action.yml]: Done in https://github.com/metanorma/ci/pull/110

File: action.yml (hunk 3 above)
Comment (CAMOBAP) [reply action.yml]: Done

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**